src/app/controllers/sap.php

```diff
@@ -3,7 +3,13 @@
 class sap extends Controller {
 
     function index() {
+        $this->load_library('token_generator_lib', 'token_generator');
+
         if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+            // Will raise E_NOTICE if csrf_token is not defined, but the library
+            // will still work fine.
+            $this->token_generator->checkCSRFToken($_POST['csrf_token']);
```

Comment:
I kept the accesses of `$_POST` in the controller, so that it's easy to pass the right name to the view too. The notice is bad, but since I want to trigger the 400 error when it's not set, I do have to call the library. I could check if `isset` and if not, pass along `NULL` or something similar.